aea/skills/base/core.py

```diff
@@ -39,15 +40,71 @@
 SkillId = str
 
 
-class Context:
+class AgentContext:
     """Save relevant data for the agent."""
 
     def __init__(self, agent_name: str, outbox: OutBox):
-        """Initialize a context object."""
-        self.agent_name = agent_name
-        self.outbox = outbox
+        """
+        Initialize a skill context.
+
+        :param agent_name: the agent's name
+        :param outbox: the outbox
+        """
+        self._agent_name = agent_name
+        self._outbox = outbox
         self.skill_loader = ConfigLoader("skill-config_schema.json", SkillConfig)
 
+    @property
+    def agent_name(self) -> str:
+        """Get agent name."""
+        return self._agent_name
+
+    @property
+    def outbox(self) -> OutBox:
+        """Get outbox."""
+        return self._outbox
+
+
+class SkillContext:
```

Comment:
Via SkillContext the handlers, tasks and behaviours have access to each other and the AgentContext.